### PR TITLE
Change FxViewData::rootNode type from Parent to Object

### LINKED ISSUE #76: @FxView does not support FXML with Stage/Dialog as its root
If we have an FXML file with Stage/Dialog/Alert as its root, constructing `FxViewRepository` will fail because Stage/Dialog/Alert do not extend `Parent` type:

https://github.com/quarkiverse/quarkus-fx/blob/85e6e00c2e89e81ad7386eeebf0e3ab3a21c3faa/runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java#L117

### PR DESCRIPTION
**Pull request by Eng-Fouad**
Fixes #76

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java [hunk 1]
```diff
@@ -155,6 +163,20 @@ private static InputStream lookupResourceAsStream(final ClassLoader classLoader,
         return stream;
     }
 
+    private static ObservableList<String> getFxmlObjectStyleSheets(Object rootNode) {
```
Comment (CodeSimcoe) [inline runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java:166]:
```suggestion
    private static ObservableList<String> getFxmlObjectStyleSheets(final Object rootNode) {
```

File: runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java [hunk 2]
```diff
@@ -155,6 +163,20 @@ private static InputStream lookupResourceAsStream(final ClassLoader classLoader,
...
+        ObservableList<String> stylesheets = null;
```
Comment (CodeSimcoe) [inline runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java:167]:
```suggestion
        ObservableList<String> stylesheets;
```

File: runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java [hunk 3]
```diff
@@ -155,6 +163,20 @@ private static InputStream lookupResourceAsStream(final ClassLoader classLoader,
...
+        if (rootNode instanceof Parent p) {
+            stylesheets = p.getStylesheets();
+        } else if (rootNode instanceof Window w) {
+            stylesheets = w.getScene().getStylesheets();
+        } else if (rootNode instanceof Scene s) {
+            stylesheets = s.getStylesheets();
+        } else if (rootNode instanceof Dialog<?> d) {
+            stylesheets = d.getDialogPane().getStylesheets();
+        }
```
Comment (CodeSimcoe) [inline runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java:176]:
```suggestion
        } else {
          stylesheets = null;
        }
```

Comment (CodeSimcoe) [reply runtime/src/main/java/io/quarkiverse/fx/views/FxViewRepository.java:176]: Java 21 would have allowed us to use pattern matching here :/

**Review by CodeSimcoe — CHANGES_REQUESTED** [review]
Hello and thanks.

Consider adding some test cases ?

**Comment by Eng-Fouad** [comment]
> Hello and thanks.
> 
> Consider adding some test cases ?

Done.

**Review by CodeSimcoe — APPROVED** [review]
Thanks !

**Comment by CodeSimcoe** [comment]
https://github.com/all-contributors add @Eng-Fouad for code

**Comment by CodeSimcoe** [comment]
@all-contributors add @Eng-Fouad for code

**Pull request merged**